provider_run_test.go

```diff
@@ -809,31 +811,37 @@ func TestPending(t *testing.T) {
 	t.Run("disallow_out_of_order", func(t *testing.T) {
 		ctx := context.Background()
 		fsys := newFsys()
-		p, err := goose.NewProvider(goose.DialectSQLite3, newDB(t), fsys,
-			goose.WithAllowOutofOrder(false),
-		)
-		check.NoError(t, err)
-		// Some migrations have been applied.
-		_, err = p.ApplyVersion(ctx, 1, true)
-		check.NoError(t, err)
-		_, err = p.ApplyVersion(ctx, 2, true)
-		check.NoError(t, err)
-		hasPending, err := p.HasPending(ctx)
-		check.NoError(t, err)
-		check.Bool(t, hasPending, true)
-		current, target, err := p.CheckPending(ctx)
-		check.NoError(t, err)
-		check.Number(t, current, 2)
-		check.Number(t, target, len(fsys))
-		_, err = p.Up(ctx)
-		check.NoError(t, err)
-		// All migrations have been applied.
-		hasPending, err = p.HasPending(ctx)
-		check.NoError(t, err)
-		check.Bool(t, hasPending, false)
-		current, target, err = p.CheckPending(ctx)
-		check.NoError(t, err)
-		check.Number(t, current, target)
+
+		run := func(t *testing.T, versionToApply int64) {
+			p, err := goose.NewProvider(goose.DialectSQLite3, newDB(t), fsys,
+				goose.WithAllowOutofOrder(false),
+			)
+			check.NoError(t, err)
+			// Some migrations have been applied.
+			_, err = p.ApplyVersion(ctx, 1, true)
+			check.NoError(t, err)
+			_, err = p.ApplyVersion(ctx, versionToApply, true)
+			check.NoError(t, err)
+			// TODO(mf): revisit the pending check behavior in addition to the HasPending
+			// method.
+			current, target, err := p.CheckPending(ctx)
+			check.NoError(t, err)
+			check.Number(t, current, versionToApply)
+			check.Number(t, target, len(fsys))
+			_, err = p.HasPending(ctx)
+			check.HasError(t, err)
+			check.Contains(t, err.Error(), "missing (out-of-order) migration")
```

Comment:
The goal was to optimize the pending check by only checking the latest migration, but this is not correct since the default behavior SHOULD BE to fail if there are missing (out-of-order) migrations. 

And we can only know if there are missing migrations by checking all of them. I wish there was a better way. 

Maybe an "ignore missing" option for those who enforce behavior in CI pipelines?